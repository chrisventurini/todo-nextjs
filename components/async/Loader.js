import { Fragment } from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

import LinearProgress from '@material-ui/core/LinearProgress';
import withStyles from '@material-ui/core/styles/withStyles';

const styles = {
    loaderContainer: {
        marginTop: '5px',
        height: '3px',
        position: 'fixed',
        width: '100%'
    }
};

export let Loader = ({ asyncCalls, classes }) => {
    let loader = (asyncCalls.inProgress) ? (<LinearProgress />) : (<Fragment />);

    return (
        <div className={classes.loaderContainer}>
            {loader}
        </div>
    );
};

Loader.propTypes = {
    asyncCalls: propTypes.object.isRequired,
    classes: propTypes.object.isRequired,
};

export const _mapState = state => {
    return { asyncCalls: state.asyncCalls }
};

Loader = withStyles(styles)(Loader);

export default connect(_mapState)(Loader);
