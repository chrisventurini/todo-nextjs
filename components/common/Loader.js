import LinearProgress from '@material-ui/core/LinearProgress';

import { withStyles } from '@material-ui/core/styles';

const styles = {
};

const Loader = ({ classes }) => {

    return (
        <LinearProgress />
    );
};

export default withStyles(styles)(Loader);