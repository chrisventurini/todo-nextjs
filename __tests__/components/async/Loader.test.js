import React from 'react';
import { shallow, mount } from 'enzyme';

import LinearProgress from '@material-ui/core/LinearProgress';

import { Loader, _mapState } from '../../../components/async/Loader';

describe('<Loader />', () => {

    describe('constructing and rendering', () => {
        let SUTWrapper,
            asyncCalls,
            stubClasses;

        describe('with async calls in progress', () => {

            beforeEach(() => {
                stubClasses = { loaderContainer: 'loaderContainer' };
                asyncCalls = { inProgress: true };

                SUTWrapper = mount(<Loader asyncCalls={asyncCalls} classes={stubClasses} />);
            });

            it('should render correctly', () => {
               expect(SUTWrapper).toMatchSnapshot();
            });

            it('should render a <LinearProgress />', () => {
                let foundEls = SUTWrapper.find(LinearProgress);

                expect(foundEls).toHaveLength(1);
            });

        });

        describe('with no async calls in progress', () => {

            beforeEach(() => {
                stubClasses = { loaderContainer: 'loaderContainer' };
                asyncCalls = { inProgress: false };

                SUTWrapper = mount(<Loader asyncCalls={asyncCalls} classes={stubClasses} />);
            });

            it('should render correctly', () => {
               expect(SUTWrapper).toMatchSnapshot();
            });

            it('should render a <LinearProgress />', () => {
                let foundEls = SUTWrapper.find(LinearProgress);

                expect(foundEls).toHaveLength(0);
            });

        });

    });

});