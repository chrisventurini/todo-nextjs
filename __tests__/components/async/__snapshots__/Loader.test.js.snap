// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> constructing and rendering with async calls in progress should render correctly 1`] = `
<WithStyles(Loader)
  asyncCalls={
    Object {
      "inProgress": true,
    }
  }
  classes={
    Object {
      "loaderContainer": "loaderContainer",
    }
  }
>
  <Loader
    asyncCalls={
      Object {
        "inProgress": true,
      }
    }
    classes={
      Object {
        "loaderContainer": "Loader-loaderContainer-1 loaderContainer",
      }
    }
  >
    <div
      className="Loader-loaderContainer-1 loaderContainer"
    >
      <WithStyles(LinearProgress)>
        <LinearProgress
          classes={
            Object {
              "bar": "MuiLinearProgress-bar-10",
              "bar1Buffer": "MuiLinearProgress-bar1Buffer-15",
              "bar1Determinate": "MuiLinearProgress-bar1Determinate-14",
              "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate-13",
              "bar2Buffer": "MuiLinearProgress-bar2Buffer-18",
              "bar2Determinate": "MuiLinearProgress-bar2Determinate-17",
              "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate-16",
              "barColorPrimary": "MuiLinearProgress-barColorPrimary-11",
              "barColorSecondary": "MuiLinearProgress-barColorSecondary-12",
              "buffer": "MuiLinearProgress-buffer-5",
              "colorPrimary": "MuiLinearProgress-colorPrimary-3",
              "colorSecondary": "MuiLinearProgress-colorSecondary-4",
              "dashed": "MuiLinearProgress-dashed-7",
              "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary-8",
              "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary-9",
              "query": "MuiLinearProgress-query-6",
              "root": "MuiLinearProgress-root-2",
            }
          }
          color="primary"
          variant="indeterminate"
        >
          <div
            className="MuiLinearProgress-root-2 MuiLinearProgress-colorPrimary-3"
            role="progressbar"
          >
            <div
              className="MuiLinearProgress-bar-10 MuiLinearProgress-barColorPrimary-11 MuiLinearProgress-bar1Indeterminate-13"
              style={Object {}}
            />
            <div
              className="MuiLinearProgress-bar-10 MuiLinearProgress-barColorPrimary-11 MuiLinearProgress-bar2Indeterminate-16"
              style={Object {}}
            />
          </div>
        </LinearProgress>
      </WithStyles(LinearProgress)>
    </div>
  </Loader>
</WithStyles(Loader)>
`;

exports[`<Loader /> constructing and rendering with no async calls in progress should render correctly 1`] = `
<WithStyles(Loader)
  asyncCalls={
    Object {
      "inProgress": false,
    }
  }
  classes={
    Object {
      "loaderContainer": "loaderContainer",
    }
  }
>
  <Loader
    asyncCalls={
      Object {
        "inProgress": false,
      }
    }
    classes={
      Object {
        "loaderContainer": "Loader-loaderContainer-1 loaderContainer",
      }
    }
  >
    <div
      className="Loader-loaderContainer-1 loaderContainer"
    />
  </Loader>
</WithStyles(Loader)>
`;
